### PR TITLE
Implement XGQ BIST test

### PR DESCRIPTION
**Pull request by chienwei-lan**
#### Problem solved by the commit
Implement XGQ based BIST test

#### What has been tested and how, request additional testing if necessary
Test xbutil validate on U50

```
Validate Device           : [0000:65:00.1]
    Platform              : xilinx_u50_gen3x16_xdma_base_4
    SC Version            : 5.2.6
    Platform ID           : 50D6E855-7F18-DEB4-1ACD-31EB231FAA9C
-------------------------------------------------------------------------------
Test 1 [0000:65:00.1]     : bist
    Details               : ERT clock frequency: 250.1 MHz
                            CQ read    4 bytes:   11 cycles
                            CQ write   4 bytes:    6 cycles
                            CU read    4 bytes:   50 cycles
                            CU write   4 bytes:   46 cycles
                            Data Integrity Test:   Pass
                            ERT sleep/wake successfully
    Details               : IOPS: 559766 (verify)
    Test Status           : [PASSED]
```

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_ctrl.c [hunk 1]
```diff
@@ -120,34 +345,89 @@ static void ert_ctrl_submit(struct kds_ert *ert, struct kds_command *xcmd)
...
 	} else {
+		ert_ctrl_post_process(ec, sq_hdr->opcode, &cq_hdr);
 		xcmd->status = KDS_COMPLETED;
 		xcmd->cb.notify_host(xcmd, xcmd->status);
 	}
 	xcmd->cb.free(xcmd);
 }
 
+
+static void ert_ctrl_submit_exit_cmd(struct ert_ctrl *ec)
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_ctrl.c:389]: why we can't make ert_ctrl_submit() work for exit cmd?

**Review by maxzhen — COMMENTED** [review]
When we do bist on u30 which does not support bist, does it error out gracefully?

**Comment by chienwei-lan** [comment]
> When we do bist on u30 which does not support bist, does it error out gracefully?

Yes, it will check the capability first

```
Starting validation for 1 devices
Validate Device           : [0000:b3:00.1]
Platform              : xilinx_u30_gen3x4_base_1
SC Version            : 6.3.8
Platform ID           : 1B5FEB2A-91B6-818D-A3E8-D9867DE17DA0

Test 1 [0000:b3:00.1]     : bist
Details               : This platform does not support ERT validate feature
Test Status           : [SKIP]
```

**Review by mamin506 — APPROVED** [review]

**Comment by dayeh-xilinx** [comment]
retest this please.

**Review by maxzhen — APPROVED** [review]

File: src/runtime_src/core/include/xgq_cmd_ert.h [hunk 2]
```diff
@@ -137,6 +137,20 @@ struct xgq_cmd_config_start {
...
+struct xgq_cmd_access_valid {
+	struct xgq_cmd_sq_hdr hdr;
+};
+
+struct xgq_cmd_data_integrity {
+	struct xgq_cmd_sq_hdr hdr;
+
+	uint32_t rw_count;
+	uint32_t draft;
+};
```
Comment (stsoe) [inline src/runtime_src/core/include/xgq_cmd_ert.h:153]: Please comment document these structs.

**Review by stsoe — CHANGES_REQUESTED** [review]
It would be great with some comments in the code.
Document the structs, functions, and code added unless it all is self-explanatory.

File: src/runtime_src/core/include/xgq_cmd_ert.h (hunk 2 above)
Comment (chienwei-lan) [reply src/runtime_src/core/include/xgq_cmd_ert.h]: > Please comment document these structs.
definition added, please help me review it, thanks!

**Review by stsoe — APPROVED** [review]

**Pull request merged**